lib/src/client/channel.dart

```diff
@@ -75,7 +78,13 @@ abstract class ClientChannelBase implements ClientChannel {
   @override
   ClientCall<Q, R> createCall<Q, R>(
       ClientMethod<Q, R> method, Stream<Q> requests, CallOptions options) {
-    final call = ClientCall(method, requests, options);
+    var call;
+    if (enableTimelineLogging) {
```

Comment:
This can potentially be changed to something like:

```dart
final call = ClientCall(method, requests, options, enableTimelineLogging ? TimelineTask(...) : null);
```

Reply:
Done